.eslintplugin.js

```diff
@@ -40,7 +40,8 @@ function getNameOfFunction(node) {
             // @ts-ignore
             return node.id.name;
         default:
-            throw new Error('Incorrect type of node has been passed to getNameOfFunction');
+            console.warn(`Warning: Incorrect type of node has been passed to getNameOfFunction: %s`, node.type);
+            return `Unknown ${(node.type || 'function')}`;
```

Comment:
This was causing the linting to fail on `CallExpression` type, and I wasn't sure how to resolve, so I thought I'd turn it into a non-fatal error.

Reply:
Oh, yeah this is a custom plugin, so I only added those expressions that I encounter within the code-base. I will have a look and see where this is being errored.

Reply:
Hmm it doesn't seem to be failing on the latest commit? Can be reverted?

Reply:
The reason this code exists, if I recall correctly, is to generate an error message for a linting error that's already been triggered. This change doesn't affect the original linting error. You still get that.

With the fatal error, eslint crashes and doesn't give you the stack trace you need to figure out the problem. All you get is "Incorrect type of node has been passed to getNameOfFunction", which is useless. It essentially means "I crashed while gathering the information you will need to fix the linting problem", but you'll still need to fix the linting problem.

Reply:
> With the fatal error, eslint crashes and doesn't give you the stack trace you need to figure out the problem. All you get is "Incorrect type of node has been passed to getNameOfFunction", which is useless. It essentially means "I crashed while gathering the information you will need to fix the linting problem", but you'll still need to fix the linting problem.

Well - if it currently doesn't warn about it than I don't consider it adding it to the function as it now will return "valid" code back(which can lead to other problems).

Reply:
Hm, I'm not sure I follow. I think you're saying you'd still like the change reverted, right? No problem.

Reply:
Removed now.